integration_tests/base_routes.py

```diff
@@ -833,33 +825,259 @@ def sync_router_di(request, router_dependencies):
+
+
+@app.get("/async/split_request_typed/headers")
+async def async_split_request_headers(request_headers: Headers):
+    return request_headers.get("server")
+
+
+@app.get("/sync/split_request_typed/path_params/:id")
+def sync_split_request_path_params(path_data: PathParams):
+    return path_data
+
+
+@app.get("/async/split_request_typed/path_params/:id")
+async def async_split_request_path_params(path_data: PathParams):
+    return path_data
+
+
+@app.get("/sync/split_request_typed/method")
+def sync_split_request_method(request_method: Method):
+    return request_method
+
+
+@app.get("/async/split_request_typed/method")
+async def async_split_request_method(request_method: Method):
+    return request_method
+
+
+@app.post("/sync/split_request_typed/body")
+def sync_split_request_body(request_body: Body):
+    return request_body
+
+
+@app.post("/async/split_request_typed/body")
+async def async_split_request_body(request_body: Body):
+    return request_body
+
+
+@app.post("/sync/split_request_typed/combined")
+def sync_split_request_combined(
+    request_body: Body,
+    query_data: QueryParams,
+    request_method: Method,
+    request_url: Url,
+    request_headers: Headers,
+):
+    return {
+        "body": request_body,
+        "query_params": query_data.to_dict(),
+        "method": request_method,
+        "url": request_url.path,
+        "headers": request_headers.get("server"),
+    }
+
+
+@app.post("/async/split_request_typed/combined")
+async def async_split_request_combined(
+    request_body: Body,
+    query_data: QueryParams,
+    request_method: Method,
+    request_url: Url,
+    request_headers: Headers,
+):
+    return {
+        "body": request_body,
+        "query_params": query_data.to_dict(),
+        "method": request_method,
+        "url": request_url.path,
+        "headers": request_headers.get("server"),
+    }
+
+
+@app.post("/sync/split_request_typed_untyped/combined")
+def sync_split_request_typed_untyped_combined(
+    query_params,
+    request_method: Method,
+    request_body: Body,
+    url: Url,
+    headers: Headers,
+):
+    return {
+        "body": request_body,
+        "query_params": query_params.to_dict(),
+        "method": request_method,
+        "url": url.path,
+        "headers": headers.get("server"),
+    }
+
+
+@app.post("/async/split_request_typed_untyped/combined")
+async def async_split_request_typed_untyped_combined(
+    query_params,
+    request_method: Method,
+    request_body: Body,
+    url: Url,
+    headers: Headers,
+):
+    return {
+        "body": request_body,
+        "query_params": query_params.to_dict(),
+        "method": request_method,
+        "url": url.path,
+        "headers": headers.get("server"),
+    }
+
+
+@app.post("/sync/split_request_typed_untyped/combined/failure")
+def sync_split_request_typed_untyped_combined_failure(query_params, request_method: Method, request_body: Body, url: Url, headers: Headers, vishnu):
```

Comment:
shouldn't everything be untyped here?

Reply:
what, why? these were the tests for using combined typed & untyped params -- to test both the annotations & keywords